cpp/Esri/ArcGISRuntime/Toolkit/GenericListModel.cpp

```diff
@@ -0,0 +1,379 @@
+    if (o)
+      disconnect(o, &QObject::destroyed, this, nullptr);
+
+    delete o;
+
+    m_objects.removeAt(i);
+  }
+  endRemoveRows();
+  return true;
+}
+
+void GenericListModel::setElementType(const QMetaObject* metaObject)
+{
+  beginResetModel();
+  for (auto o : m_objects)
+  {
+    // Ensure additional removal signals are not triggered.
+    if (o)
+      disconnect(o, &QObject::destroyed, this, nullptr);
+
+    delete o;
+  }
+  m_objects.clear();
+  m_elementType = metaObject;
+  m_displayPropIndex = -1;
+  endResetModel();
+}
+
+const QMetaObject* GenericListModel::elementType() const
+{
+  return m_elementType;
+}
+
+void GenericListModel::setDisplayPropertyName(const QString& propertyName)
+{
+  m_displayPropIndex = m_elementType->indexOfProperty(propertyName.toLatin1());
+}
+
+QString GenericListModel::displayPropertyName()
+{
+  if (m_displayPropIndex < 0)
+    return "";
+  else
+    return m_elementType->property(m_displayPropIndex).name();
+}
+
+bool GenericListModel::append(QObject* object)
+{
+  if (!m_elementType)
+    return false;
+  
+  if (!object)
+    return false;
+
+  if (object->metaObject() != m_elementType)
+    return false;
+
+  auto i = rowCount();
+  beginInsertRows(QModelIndex(), i, i);
+  m_objects << object;
+  endInsertRows();
+  connectElement(index(i));
+  return true;
+}
+
+bool GenericListModel::append(QList<QObject*> objects)
+{
+  const auto size = objects.size();
+
+  if (!m_elementType)
+    return false;
+  if (size < 1)
+    return true;
+
+  for (auto o: objects)
+  {
+    if (!o)
+      return false;
+    else if (o->metaObject() != m_elementType)
+      return false;
+  }
+
+  auto i = rowCount();
+
+  beginInsertRows(QModelIndex(), i, i + size - 1);
+  m_objects << objects;
+  endInsertRows();
+
+  for (; i < size; ++i)
+    connectElement(index(i));
+
+  return true;
+}
+
+void GenericListModel::connectElement(QModelIndex index)
+{
+  if (!m_elementType)
+    return;
+  else if (!index.isValid())
+    return;
+
+  QObject* object = m_objects.at(index.row());
+
+  if (!object || object->metaObject() != m_elementType)
+    return;
+
+  // If object is deleted externally we remove from the model.
+  connect(object, &QObject::destroyed, this,
+    [this, pIndex = QPersistentModelIndex(index)]
+    {
+      if (!pIndex.isValid())
+        return;
+
+      const auto row = pIndex.row();
+      
+      if (row < 0 || row >= m_objects.size())
+        return;
+
+      m_objects[row] = nullptr; // Prevents double delete.
+      removeRow(row, pIndex.parent());
+    }
+  );
+
+  // Connect to each property notifySignal and hook up to our dataChanged signal
+  // using MetaElement objects as drop-in replacements for lambdas.
+  const auto offset = m_elementType->propertyOffset();
+  for (int i = offset; i < m_elementType->propertyCount(); ++i)
+  {
+    auto property = m_elementType->property(i);
+    auto notifySignal = property.notifySignal();
+    if (notifySignal.isValid())
```

Comment:
Would https://doc.qt.io/qt-5/qmetaproperty.html#hasNotifySignal be clearer?

Reply:
Good find! Yes it would be!